Sources/SwiftCliCore/MetaDataBuilder .swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+
+public struct RowInfo: Equatable {
+    var row: [String]
+    var rowNumber: Int
+    var bottomLine: String = ""
+    var stitchSymbols: String = ""
+    var width: Int = 0
+    var patternRowsCount: Int = 0
+    var leftIncDec: Int = 0
+    var rightIncDec: Int = 0
+    // This is about the number of "empty stitches" that should be added on the left to account for future increases
+    var leftOffset: Int = 0
+    var transRowLeftOffset: Int = 0
+    lazy var leftOffsetString: String = String(repeating: "  ", count: leftOffset)
+    lazy var transRowLeftOffsetString: String = String(repeating: "  ", count: transRowLeftOffset)
+    lazy var offsetBottomLine: String = leftOffsetString + bottomLine
+    lazy var offsetStitchSymbols: String = transRowLeftOffsetString + leftOffsetString + stitchSymbols
+    lazy var totalRow: String = offsetStitchSymbols + offsetBottomLine
+
+}
+public class MetaDataBuilder {
+    public init() {}
+
+    public func gatherAllMetaData(stitchArray: [[String]]) -> [RowInfo] {
+        var allRowsMetaData = [] as [RowInfo]
+        let rowsNum = stitchArray.count
+        var upcomingleftOffset = 0
+        for row in 0..<(rowsNum) {
+            var newRowMetadata = makeRowMetadata(stitchRow: stitchArray[row], rowNumber: row)
+            newRowMetadata.leftOffset += upcomingleftOffset
+            if newRowMetadata.leftIncDec > 0 && row > 0 {
+                for prevRow in (0...(row-1)).reversed() {
+                    allRowsMetaData[prevRow].leftOffset += newRowMetadata.leftIncDec
+                }
+            }
+            if newRowMetadata.leftIncDec < 0 {
+                upcomingleftOffset += abs(newRowMetadata.leftIncDec)
+                newRowMetadata.transRowLeftOffset += upcomingleftOffset
+            }
+            allRowsMetaData.append(newRowMetadata)
+        }
+        return(allRowsMetaData)
+    }
+
+    public func makeRowMetadata(stitchRow: [String], rowNumber: Int) -> RowInfo {
+        var rowData = RowInfo(row: stitchRow, rowNumber: rowNumber)
+        rowData.leftIncDec = findLeftChanges(stitchRow: stitchRow)
+        rowData.rightIncDec = findRightChanges(stitchRow: stitchRow)
+        rowData.width = stitchRow.count
+        if rowNumber == 0 {
+            rowData.bottomLine = ChartConstructor().makeBottomRow(width: rowData.width)
+        } else {
+            rowData.bottomLine = ChartConstructor().makeMiddleRow(width: rowData.width,
```

Comment:
If the parameters for a function call won't fit on one line, then you should put a newline in between the braces and the first parameter:

```swift
ChartConstructor().makeMiddleRow(
    width: rowData.width,
    rDiff: rowData.rightIncDec,
    lDiff: rowData.leftIncDec
)
```